pkg/transformers/openshift/openshift.go

```diff
@@ -0,0 +1,655 @@
+package openshift
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilerrs "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+	"github.com/openshift/origin/pkg/generate/git"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	dockerfileutil "github.com/openshift/origin/pkg/util/docker/dockerfile"
+	"github.com/openshift/origin/third_party/github.com/docker/libcompose/project"
+
+	// Install OpenShift APIs
+	_ "github.com/openshift/origin/pkg/build/api/install"
+	_ "github.com/openshift/origin/pkg/deploy/api/install"
+	_ "github.com/openshift/origin/pkg/image/api/install"
+	_ "github.com/openshift/origin/pkg/route/api/install"
+	_ "github.com/openshift/origin/pkg/template/api/install"
+)
+
+type TransformData struct {
+	errs         []error
+	warnings     map[string][]string
+	serviceOrder sets.String
+	joins        map[string]sets.String
+	volumesFrom  map[string][]string
+	colocated    []sets.String
+	builds       map[string]*app.Pipeline
+	pipelines    app.PipelineGroup
+	containers   map[string]*kapi.Container
+	objects      app.Objects
+	bases        []string
+	aliases      map[string]sets.String
+}
+
+func Transform(p *project.Project, bases []string) error {
+	data := TransformData{
+		errs:         []error{},
+		warnings:     make(map[string][]string),
+		serviceOrder: sets.NewString(),
+		joins:        make(map[string]sets.String),
+		volumesFrom:  make(map[string][]string),
+		colocated:    []sets.String{},
+		builds:       make(map[string]*app.Pipeline),
+		pipelines:    app.PipelineGroup{},
+		containers:   make(map[string]*kapi.Container),
+		objects:      app.Objects{},
+		bases:        bases,
+		aliases:      make(map[string]sets.String),
+	}
+
+	updateServiceOrder(p, &data)
+	updateVolumes(p, &data)
+
+	err := updatePodsList(p, &data)
+	if err != nil {
+		return err
+	}
+
+	updateAliases(p, &data)
+
+	g := app.NewImageRefGenerator()
+
+	err = updateBuildPipeline(p, g, &data)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("After upateBuildPipeline")
+	fmt.Println("pipelines: ", data.pipelines)
+	fmt.Println("builds: ", data.builds)
+
+	err = updateDeploymentConfigs(p, g, &data)
+	if err != nil {
+		return err
+	}
+
+	if len(data.errs) > 0 {
+		return utilerrs.NewAggregate(data.errs)
+	}
+
+	err = updatePipelineObjects(&data)
+	if err != nil {
+		return err
+	}
+
+	err = updateServiceObjects(&data)
+
+	template := &templateapi.Template{}
```

Comment:
But if we are going to to contribute this back to Origin, we we should leave it as Template